pkp/pkp-lib#1663 anonymous default file names in review stage

bozana opened the pull request:
S. https://github.com/pkp/pkp-lib/issues/1663

asmecher commented on classes/submission/SubmissionFile.inc.php:
```diff
@@ -570,17 +570,19 @@ function _generateFileName() {
 
 	/**
 	 * Generate a user-facing name for the file
+	 * @param $anonymous boolen Whether the user name should be excluded
```
*boolean

asmecher commented on locale/en_US/common.xml:
```diff
@@ -142,6 +142,7 @@
 	<message key="common.false">False</message>
 	<message key="common.file">File</message>
 	<message key="common.file.namingPattern">{$username}, {$userGroup}, {$originalFilename}</message>
+	<message key="common.file.anonymousNamingPattern">{$userGroup}, {$originalFilename}</message>
```
Could you duplicate this throughout the other locales? Better to have it present (and accurate for most locales) than to have `##common.file.anonymousNamingPattern##` showing up as a universal default file title :)

asmecher commented on classes/submission/SubmissionFileDAODelegate.inc.php:
```diff
@@ -89,8 +89,19 @@ function insertObject($submissionFile, $sourceFile, $isUpload = false) {
 			$submissionFile->setFileId($this->_getInsertId('submission_files', 'file_id'));
 		}
 
+		$reviewStage = $submissionFile->getFileStage() == SUBMISSION_FILE_REVIEW_FILE ||
```
$reviewStage = in_array($submissionFile->getFileStage(), array(
        SUBMISSION_FILE_REVIEW_FILE, SUBMISSION_FILE_REVIEW_ATTACHMENT, SUBMISSION_FILE_REVIEW_REVISION,
    ));

asmecher replied:
(slight preference :)

asmecher commented:
I just thought of something -- this covers new uploads, but what about detected revisions of existing uploads (a common case), or files directly promoted forward, which will already have the author's name in the title? The only easy way to cover both of those, I think, is to just go with the user group name throughout and remove the author name entirely. What do you think, @bozana?

bozana commented:
Hmmm... I tested it and what I could figure out is that on each place at the review stage only the review stage files are provided for selection, in review files grid, in reviewers grid, new review round, in revisions grid, when requesting revisions and asking for resubmission for review. And those are all anonymized, if they have default names. Thus it seems to be safe/OK, but I am not 100% sure if I oversee something...

asmecher commented:
Isn't it possible to promote files forward into the review files grid when sending them from the submission stage, when recording a "send to review" decision?

bozana commented:
Yes, but that I havel covered, within the second l se statement, when the file name exists and has default nam. Thus those are anonymized. (From in between flights :-))

bozana commented:
I've just rebased and added your suggestions. I will leave the decision to you, if this way or everywhere/everything anonymous. In that second case, I think just the current locale key will have to be changed and eventually copied everywhere.

asmecher commented:
Thanks, @bozana! Have a great vacation.
